src/auth/negotiate/wrapper/negotiate_wrapper.cc

```diff
@@ -257,6 +257,7 @@ processingLoop(FILE *FDKIN, FILE *FDKOUT, FILE *FDNIN, FILE *FDNOUT)
                 return 0;
             }
         }
+        buff[sizeof(buff)-1] = '\0';
```

Comment:
Like you said, the official code probably does terminate `buff` correctly so this proposed code causes no harm, but if we ignore the well-hidden "already terminated correctly" fact, then this proposed code looks as if it silently truncates input! We can avoid that (false) implication by doing something like this:

```suggestion
        buff[sizeof(buff)-1] = '\0';
        assert(strlen(buff) < sizeof(buf)-1); // the previous assignment did not truncate
```

or this


```suggestion
        buff[sizeof(buff)-1] = '\0'; // paranoid; already terminated correctly
```

I do not insist on this change.

Reply:
Thank you for addressing this concern.